commit 813ec8d137cac68746ed38fc07693663a13bc73c (refs/changes/79/279/1)
Author: Marisa Emerson <mje@uq.edu.au>
Date:   2016-08-12T17:46:03+10:00 (3 years, 2 months ago)
    
    joyent/node-triton-tags#2 allow dns names to start with a number

diff --git a/lib/cns-svc-tag.pegjs b/lib/cns-svc-tag.pegjs
index 09df8a5..83b5b2a 100644
--- a/lib/cns-svc-tag.pegjs
+++ b/lib/cns-svc-tag.pegjs
@@ -14,7 +14,7 @@ service = name:dnslabel port:(":" int)? props:(":" key "=" value)*
 	});
 	return (svc);
 }
-dnslabel "DNS name"		= [a-zA-Z] [a-zA-Z0-9-.]* {
+dnslabel "DNS name"		= [a-zA-Z0-9] [a-zA-Z0-9-.]* {
 	return (text().toLowerCase());
 }
 key "property name"		= [a-z] [a-z0-9-]* 	   { return (text()); }
diff --git a/test/unit.test.js b/test/unit.test.js
index 8e34b42..577b4d5 100644
--- a/test/unit.test.js
+++ b/test/unit.test.js
@@ -176,6 +176,11 @@ var cases = [
         key: 'triton.cns.services',
         str: 'foobar:priority=aaaaaaa',
         err: /must be within the range 0 - 65535/
+    },
+    {
+        key: 'triton.cns.services',
+        str: '5foobar',
+        val: '5foobar'
     }
 
     // TODO: triton.cns.reverse_ptr
